// ref: https://knowledge.sakura.ad.jp/24059/

syntax = "proto3";

// Request and Response
message Request{ string json_str = 1; }
message Response{ string json_str = 2; }

// FRS
service FRSPropagation { rpc on_post(Request) returns (Response){} }
service FRSTermination { rpc on_post(Request) returns (Response){} }
service Lock { rpc on_post(Request) returns (Response){} }
service Unlock { rpc on_post(Request) returns (Response){} }

// 2PL
service TPLPropagation { rpc on_post(Request) returns (Response){} }
service TPLTermination { rpc on_post(Request) returns (Response){} }

// common
service YCSBLoad { rpc on_get(Request) returns (Response){} }
service YCSB { rpc on_get(Request) returns (Response){} }
service TPCCLoadLocal { rpc on_get(Request) returns (Response){} }
service TPCCLoadCustomer { rpc on_get(Request) returns (Response){} }
service TPCC { rpc on_get(Request) returns (Response){} }
service ValChange { rpc on_get(Request) returns (Response){} }


// python3 -m grpc_tools.protoc -I＜プロトコル定義ファイルが格納されているディレクトリ＞ --python_out=＜コード出力先ディレクトリ＞ --grpc_python_out=＜コード出力先ディレクトリ＞ ＜プロトコル定義ファイルのパス名＞
// python3 -m grpc_tools.protoc -I . --python_out=. --grpc_python_out=. data.proto